collector/src/toolchain.rs

```diff
@@ -400,6 +405,26 @@ pub fn get_local_toolchain(
             None
         };
 
+        let clippy =
+        if let Some(clippy) = &clippy {
+            Some(clippy.canonicalize().with_context(|| {
+                format!("failed to canonicalize rustdoc executable {:?}", clippy)
+            })?)
+        } else if profiles.contains(&Profile::Clippy) {
+            // We need a `clippy`. Look for one next to `rustc`.
+            if let Ok(clippy) = rustc.with_file_name("clippy").canonicalize() {
```

Comment:
I think that this should be `cargo-clippy` instead of `clippy`. It is called `cargo-clippy` in my local toolchains installed by `rustup`.